### PR TITLE
Capped the number of enemies

### PR DESCRIPTION
**Pull request by Toe-Knee-Y**
Will write the wave now, figured out how each wave works

### CODE REVIEW

File: Assets/Scripts/EnemySpawn.cs [hunk 1]
```diff
@@ -3,30 +3,108 @@
...
             _cooldownDelay = SpeedManager.enemySpawnScaling;
-            SpawnEnemy();
-            _cooldown = true;
-            StartCoroutine(Cooldown());
+            if (canSpawn && currentNumEnemies < enemiesPerWave)
+            {
+                SpawnEnemy();
+                _cooldown = true;
+                currentNumEnemies = GameObject.FindGameObjectsWithTag("Enemy").Length;
+                Debug.Log(currentNumEnemies);
```
Comment (Sirkular) [inline Assets/Scripts/EnemySpawn.cs:87]: REMOVE.

**Review by Sirkular — APPROVED** [review]

File: Assets/Scripts/EnemySpawn.cs [hunk 2]
```diff
@@ -3,30 +3,108 @@
...
+                StartCoroutine(Cooldown());
+            }
+            else
+            {
+                if(currentNumEnemies >= enemiesPerWave)
+                {
+                    canSpawn = false;
+                    Debug.Log(GameObject.FindGameObjectsWithTag("Enemy").Length);
```
Comment (zhuandy42) [inline Assets/Scripts/EnemySpawn.cs:95]: we do be debugging

File: .vsconfig [hunk 3]
```diff
@@ -0,0 +1,6 @@
+﻿{
+  "version": "1.0",
+  "components": [
+    "Microsoft.VisualStudio.Workload.ManagedGame"
+  ]
+}
```
Comment (MaXeraph) [inline .vsconfig:6]: REMOVE.

File: Assets/Scripts/EnemySpawn.cs [hunk 4]
```diff
@@ -3,30 +3,108 @@
...
+    private bool canSpawn = false;
+    private bool startSpawn = false;
+    private float nextSpawnTime;
+    private Wave currentWave;
+    private int currentWaveNumber = 0;
+
+    private const int totalWaveNumber = 3;
+    private const int enemiesPerWave = 10;
+    private const int spawnInterval = 0;
+    private int currentNumEnemies = 0;
```
Comment (MaXeraph) [inline Assets/Scripts/EnemySpawn.cs:39]: dont have to put `private`, variables are by default private

Comment (Toe-Knee-Y) [reply Assets/Scripts/EnemySpawn.cs:39]: thank you max

**Pull request merged**